models/items.js

```diff
@@ -14,24 +12,26 @@ const addTagsToItem = async (itemData) => {
   try {
     const { itemId, itemType, tagPayload } = itemData;
     const batch = firestore.batch();
+    let isNewtag = false;
     for (const tag of tagPayload) {
-      const tagData = await tagModel.doc(tag.tagId).get();
-      const levelData = await levelModel.doc(tag.levelId).get();
+      const itemData = await itemTagsModel
+        .where("itemId", "==", itemId)
+        .where("tagId", "==", tag.tagId)
+        .where("levelId", "==", tag.levelId)
+        .get();
+      if (!itemData.empty) continue;
+      isNewtag = true;
       const itemTag = {
         itemId,
         itemType: itemType.toUpperCase(),
         tagId: tag.tagId,
-        tagName: tagData.data().name,
-        tagType: tagData.data().type,
```

Comment:
I guess the tag type should exist. It will help in filtering in future to fetch users with specific tag and tasks with specific tags

Reply:
I have added fetching the tag data only based on ID according to the current requirements. We can discuss this with Ankush.